### PR TITLE
Remove unused fixture parameters

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR removes most unused fixture parameters from the tests, found with `vulture`. There were plenty more, but those are fixtures which perform side effects and are required for the tests to pass https://github.com/Clinical-Genomics/cg/issues/2063.

### Fixed
- Remove unused fixture parameters

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/cli/workflow/balsamic/test_store_housekeeper.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def test_without_config(cli_runner: CliRunner, balsamic_context: CGConfig, caplo
 
 
 def test_case_without_deliverables_file(
-    cli_runner: CliRunner, balsamic_context: CGConfig, mock_config: dict, caplog
+    cli_runner: CliRunner, balsamic_context: CGConfig, mock_config, caplog
```
Comment (henrikstranneheim) [inline tests/cli/workflow/balsamic/test_store_housekeeper.py:69]: Was the type hint wrong or could we also remove the parameter?

**Review by henrikstranneheim — APPROVED** [review]
Nice to get rid of these fixture parameters 👍

File: tests/cli/workflow/balsamic/test_store_housekeeper.py (hunk 1 above)
Comment (seallard) [reply tests/cli/workflow/balsamic/test_store_housekeeper.py:69]: Yes, the type hint was wrong. The fixture does not return anything and does some setup, the test fails without it. I created an issue for it: https://github.com/Clinical-Genomics/cg/issues/2063

**Review by diitaz93 — APPROVED** [review]
Another great initiative 👍

**Pull request merged**